cookbooks/bcpc-hadoop/recipes/zabbix_web_monitoring.rb

```diff
@@ -0,0 +1,135 @@
+ruby_block "zabbix_web_monitoring" do
+  block do
+    require 'zabbixapi'
+
+    # Make connection to zabbix api url
+    zbx = ZabbixApi.connect(
+      :url => "https://#{node['bcpc']['management']['vip']}" +
+        ":#{node['bcpc']['zabbix']['web_port']}/api_jsonrpc.php",
+      :user => get_config!('zabbix-admin-user'),
+      :password => "#{get_config!('password','zabbix-admin','os')}"
+    )
+    if zbx.nil?
+      Chef::Log.error("Could not connect to Zabbix server")
+      raise "Could not connect to Zabbix server"
+    end
+
+    # Create zabbix host group same as the chef environment name
+    hostgroup_id = zbx.hostgroups.get_id(:name => "#{node.chef_environment}")
```

Comment:
If host group is the "group" displayed in the Zabbix web UI, perhaps these groups could be more meaningful for humans, e.g. "<chef environment> <service> <daemon>" (i.e. "cluster1 monitoring graphite-daemon")?